infrastructure/terraform/modules/functions/functionassistant/function_app.py

```diff
@@ -54,28 +59,33 @@ def downloadfile(httpbody: str):
         message = f"An error occurred reading the body: {e}"
         logging.info(message)
         return message
-    
+
     openai_client = AzureOpenAI(
-        azure_ad_token_provider=token_provider,  
+        azure_ad_token_provider=token_provider,
         api_version="2024-02-15-preview",
         azure_endpoint = f"{azure_endpoint}"
     )
 
-        # Create client
+    # Create client
+    #logging.info("Creating BlobServiceClient: " + {storage_domain_name}  )
     blob_service_client = BlobServiceClient(
         storage_domain_name, credential=credential
     )
+    #logging.info("Success Creating BlobServiceClient: " + {storage_domain_name}  )
+
+    #logging.info("Creating storage_container_name: " + {storage_container_name}  )
     blob_client = blob_service_client.get_blob_client(
         container=storage_container_name, blob=storage_blob_name
     )
+    #logging.info("Success Creating storage_container_name: " + {storage_domain_name}  )
 
     try:
     # Download blob
         with open(file=f"/tmp/input_{storage_blob_name}_prompt.json", mode="wb") as transcript:
             download_stream = blob_client.download_blob()
             data = download_stream.readall()
             transcript.write(data)
-        logging.info("File downloaded successfully.")
+        #logging.info("File downloaded successfully. Blob name" + {storage_blob_name}  )
 
     except Exception as e:
         message = (f"An error occurred downloading the file: {e}, {e.args}")

```

Comment:
We need to add retrieval tool to the assistant

Reply:
Line #146 using time.sleep within an orchestrator function is not recommended as it can cause unnecessary delays and increased costs. Instead, use create_timer for durable functions.

Reply:
will be tracked as feature enhancement https://github.com/Azure/ai-hub/issues/139